rpc/resource/resource.go

```diff
@@ -0,0 +1,102 @@
+package resource
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/infobloxopen/atlas-app-toolkit/rpc/resource/resourcepb"
+)
+
+const externalResource = "<external>"
+
+var (
+	mu       sync.RWMutex
+	registry = make(map[string]*struct {
+		Manager
+	})
+)
+
+type Identifier interface {
+	fmt.Stringer
+	sql.Scanner
+	driver.Valuer
+
+	ApplicationName() string
+	ResourceType() string
+	ResourceID() string
+
+	IsExternal() bool
```

Comment:
I do really want a feedback on `Identifier` interface. Is not it overloaded?

Reply:
It seems that we could narrow interface by eliminating ‘sql.Scanner’, ‘fmt.Stringer’ and ‘driver.Valuer’ interfaces, thereby allow an user to decide is an Identifier is storable or not.

Reply:
I am fine with both ways. Probably the idea to eliminate  `sql.Scanner`, `fmt.Stringer` and `driver.Valuer` interfaces makes sence as it might be some service without DB at all.